include/deal.II/grid/manifold_lib.h

```diff
@@ -503,6 +503,150 @@ class TorusManifold : public ChartManifold<dim,3,3>
   double r, R;
 };
 
+
+
+/**
+ * A mapping class that smoothly transitions from different manifold
+ * descriptions on different faces by a so-called transfinite
+ * interpolation. As opposed to setting the manifold description only on the
+ * boundary of a mesh, this approach makes sure that the deformation of the
+ * mesh according to a curved boundary description is applied to more than one
+ * element layer once the mesh gets refined, which is necessary to get
+ * high-order convergence rates (without this class, the error on cells close
+ * to the boundary does not exceed 3 on a circle).
+ *
+ * @note Only instantiated for dim > 1 as a transfinite mapping cannot be
+ * defined in 1D.
+ *
+ * @author Luca Heltai, Martin Kronbichler, 2017
+ */
+template <int dim, int spacedim=dim>
+class TransfiniteInterpolationManifold : public Manifold<dim,spacedim>
+{
+public:
+  /**
+   * Constructor.
+   */
+  TransfiniteInterpolationManifold();
+
+  /**
+   * Initializes the manifold with a coarse mesh. The prerequisite for using
+   * this class is that the input triangulation is uniformly refined and the
+   * manifold is later attached to the same triangulation.
+   *
+   * @note For performance and accuracy reasons, it is recommended to apply
+   * the transfinite manifold to as coarse meshes as possible. Regarding accurcay, the curved
+   * description can only be applied to new points created from a given
+   * surrounding, and the grid quality is typically higher when extending the
+   * curved description over as large a domain as possible. When it comes to
+   * performance, the identification of the correct coarse cell in the
+   * get_new_point() method is generally expensive and needs to pass all
+   * coarse cells. Thus, in the current implementation this is only really
+   * economical when there is not more than a few dozen coarse cells.
+   */
+  void initialize (const Triangulation<dim,spacedim> &triangulation);
+
+  /**
+   * Return the point which shall become the new vertex surrounded by the
+   * given points @p surrounding_points. @p weights contains appropriate
+   * weights for the surrounding points according to which the manifold
+   * determines the new point's position.
+   *
+   * The implementation in this class overrides the method in the base class
```

Comment:
It actually has, see `manifold.cc:87`. The implementation calls `get_intermediate_point` which in turn calls `project_to_manifold`. The only function without implementation is `project_to_manifold`, but we bypass that function here and I would also not know what to do at this point.